### ISSUE TITLE
Why does che override the container entrypoint in a devfile on Red Hat Dev Spaces

### ISSUE THREAD

**Issue by Charleeboyle** [eclipse/che#22182]
### Summary

I am trying to migrate from RedHat Code Ready Workspaces (CRW) to Red Hat Dev Spaces. Our existing devfile for CRW does not work in Dev Spaces because Che is contributing many different fields to the devfile upon creation of the workspace.

The following attributes are added to the devfile:
```
che-code.eclipse.org/contribute-cpuLimit: true
      che-code.eclipse.org/contribute-cpuRequest: true
      che-code.eclipse.org/contribute-endpoint/che-code: 3100
      che-code.eclipse.org/contribute-endpoint/code-redirect-1: 13131
      che-code.eclipse.org/contribute-endpoint/code-redirect-2: 13132
      che-code.eclipse.org/contribute-endpoint/code-redirect-3: 13133
      che-code.eclipse.org/contribute-entry-point: true
      che-code.eclipse.org/contribute-memoryLimit: true
      che-code.eclipse.org/contribute-memoryRequest: true
      che-code.eclipse.org/contribute-volume-mount/checode: /checode
```

The part causing the most confusion is  this entrypoint command which is added to the container component of the devfile:
```
command:
        - /checode/entrypoint-volume.sh
```

From reading the devfile.io documentation, the `command` field modifies the entrypoint command of the container created from the image, and if not specified, the entrypoint from the provided image is used.

We provide a custom image for this container component, therefore when Che adds these fields into the devfile, the container does not start as intended. When this entrypoint command is removed from the devfile after the workspace is created, by editing the devfile, our container will start as intended, but then the workspace does not open.

What is the purpose of these Che contributions ? I notice that these are added to the "Empty Workspace" sample also.

### Relevant information

The devfile we want to use:
```
schemaVersion: 2.1.0
metadata:
  name: sample-workspace
projects:
  - name: sample
    git:
      checkoutFrom:
        revision: main
      remotes:
        origin: < git repo containing our project source >
components:
    - name: ui
      container:
        image: < web application image which is hosted on the 9080 port>
        memoryLimit: 4927M
        mountSources: true
        endpoints:
          - name: ui
            targetPort: 9080
            exposure: public
            protocol: http
```

We are aiming for a workspace which opens our project source, and runs the specified image in a container `ui`, exposing it on the defined endpoint.

Our image entrypoint is being overridden when the workspace is created
```
command:
        - /checode/entrypoint-volume.sh
```

Do we need two container components in our devfile? One which will host the workspace, and one which will run our custom image?

**Comment by RomanNikitenko** [comment]
Hello, @Charleeboyle 
About 
<img width="815" alt="image" src="https://user-images.githubusercontent.com/5676062/234929940-2456e811-636f-4cc6-a9e3-745d9a8f2a36.png">

I have no info about version you are using, but recently the corresponding was reworked: https://github.com/eclipse-che/che-plugin-registry/pull/1566/files

So, it should be possible to start IDE on the post-start event, the entrypoint  should not be overridden.

**Comment by Charleeboyle** [comment]
Hi @RomanNikitenko thanks for the speedy response.

Currently we are using the `Red Hat OpenShift Dev Spaces Operator version 3.5.0+0.1679521107.p provided by Red Hat` as part of the `registry.redhat.io/redhat/redhat-operator-index:v4.12` catalog source. May it be that the changes in #21738 are in a newer version of the index ?

**Comment by RomanNikitenko** [comment]
The changes will be available within `3.6` version

**Comment by Charleeboyle** [comment]
Is there a timeline for the 3.6 release ?

**Comment by RomanNikitenko** [comment]
May 17 is date of the 3.6 release if I'm not mistaken.
The new version should be available [here](https://developers.redhat.com/products/openshift-dev-spaces/download) when the release is done.

**Comment by Charleeboyle** [comment]
This issue was resolved in version 3.6. Thank you.
